docs/static/transforming-data.asciidoc

```diff
@@ -0,0 +1,520 @@
+----------------------------------
+input {
+  kafka {
+    codec => {
+      avro => {
+        schema_uri => "/tmp/schema.avsc"
+      }
+    }
+  }
+}
+...
+----------------------------------
+
+
+<<plugins-filters-csv,csv filter>>::
+
+Parses comma-separated value data into individual fields. By default, the
+filter autogenerates field names (column1, column2, and so on), or you can specify
+a list of names. You can also change the column separator.
++
+The following config parses CSV data into the field names specified in the
+`columns` field:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  csv {
+    separator => ","
+    columns => [ "Transaction Number", "Date", "Description", "Amount Debit", "Amount Credit", "Balance" ]
+  }
+}
+--------------------------------------------------------------------------------
+
+<<plugins-codecs-fluent,fluent codec>>::
+        
+Reads the Fluentd `msgpack` schema.
++
+The following config decodes logs received from `fluent-logger-ruby`:
++
+[source,json]
+--------------------------------------------------------------------------------
+input {
+  tcp {
+    codec => fluent
+    port => 4000
+  }
+}
+--------------------------------------------------------------------------------
+
+
+//ALVIN: SHOULD WE ALSO COVER THE MSGPACK CODEC? I LOOKED AT THE DOCS FOR MSGPACK AND THERE'S NO USEFUL INFO IN THERE. 
+
+<<plugins-codecs-json,json codec>>::
+
+Decodes (via inputs) and encodes (via outputs) JSON formatted content, creating
+one event per element in a JSON array.
++
+The following config decodes the JSON formatted content in a file:
++
+[source,json]
+--------------------------------------------------------------------------------
+input {
+  file {
+    path => "/path/to/myfile.json"
+    codec =>"json"
+}
+--------------------------------------------------------------------------------
+
+
+//ALVIN: SHOULD WE INCLUDE JSON_LINES HERE, TOO, OR MAYBE MENTION IT?
+
+<<plugins-codecs-protobuf,protobuf codec>>::
+
+Reads protobuf encoded messages and converts them to Logstash events. Requires
+the protobuf definitions to be compiled as Ruby files.
++
+The following config decodes events from a Kafka stream:
++
+[source,json]
+--------------------------------------------------------------------------------
+kafka {
+  zk_connect => "127.0.0.1"
+  topic_id => "your_topic_goes_here"
+  codec => protobuf {
+    class_name => "Animal::Unicorn"
+    include_path => ['/path/to/protobuf/definitions/UnicornProtobuf.pb.rb']
+  }
+}
+--------------------------------------------------------------------------------
+
+
+//REVIEWERS: PLEASE DOUBLE-CHECK THE SYNTAX FOR THE PROTOBUF EXAMPLE. I CHANGED THE FORMATTING TO BE CONSISTENT WITH THE REST OF THE DOCUMENTATION HERE, BUT DIDN'T TEST THIS EXAMPLE.
+
+<<plugins-filters-xml,xml filter>>::
+
+Parses XML into fields.
++
+The following config parses the whole XML document stored in the `message` field:
```

Comment:
> For some of the examples, I started adding the input message and result because I thought it was hard to understand 

I agree with your reasoning. For XML I agree the input may not need to be shown.